app/scripts/views/NodeViews/ThreeCSG.js

```diff
@@ -13,9 +13,7 @@ define(['backbone', 'underscore', 'jquery', 'BaseNodeView'], function(Backbone,
       BaseNodeView.prototype.initialize.apply(this, arguments);
 
       this.model.on('change:selected', this.colorSelected, this);
-      this.model.on('change:visible', this.changeVisibility, this);
       this.model.on('remove', this.onRemove, this);
-      this.model.workspace.on('change:current', this.changeVisibility, this);
```

Comment:
Creator (Workspace View) will make a node view to subscribe on these events and pass into it the workspace because the workspace is used in `changeVisibility` method